src/Analyzer.Core.BuiltInRuleTests/TestRunner.cs

```diff
@@ -30,35 +30,38 @@ public static void Initialize(TestContext testContext)
+            // Verify test config is valid (decided in GetTests function below).
+            // Reason for failure is described in RuleId.
+            if (ruleExpectations.RuleId.StartsWith("Invalid test"))
+                Assert.Fail(ruleExpectations.RuleId);
+
             // Get template to analyze
-            var testTemplatePath = Path.Combine("TestTemplates", $"{ruleExpectations.Template}.badtemplate");
-            var testTemplate = File.ReadAllText(testTemplatePath);
+            Assert.IsTrue(File.Exists(ruleExpectations.Template), message: $"Error: {ruleExpectations.Template} does not exist");
+            var testTemplate = File.ReadAllText(ruleExpectations.Template);
 
-            // If not already analyzed, analyze it and store evaluations
-            if (!templateEvaluations.TryGetValue(testTemplatePath, out var evaluations))
-            {
-                evaluations = templateAnalyzer.AnalyzeTemplate(testTemplate);
-                templateEvaluations[testTemplatePath] = evaluations;
-            }
+            // Analyze template
+            var evaluations = templateAnalyzer.AnalyzeTemplate(testTemplate);
 
             // Find any instances of the rule being tested
-            // Exception containing "Sequence contains no elements" likely means you did 
-            // not name the test file the same name as the rule
-            var thisRuleEvaluations = evaluations.Where(e => e.RuleId.Equals(ruleExpectations.TestName, StringComparison.OrdinalIgnoreCase)).ToList();
+            var thisRuleEvaluations = evaluations.Where(e => e.RuleId.Equals(ruleExpectations.RuleId, StringComparison.OrdinalIgnoreCase)).ToList();
 
             // Get all lines reported as failed
             var failingLines = thisRuleEvaluations
                 .Where(e => !e.Passed)
                 .SelectMany(e => GetAllFailedLines(e))
-                .ToHashSet();
+                .ToList();
+
+            failingLines.Sort();
 
             // Verify all expected lines are reported
-            var expectedLines = ruleExpectations.ReportedFailures.Select(failure => failure.LineNumber).ToHashSet();
-            Assert.IsTrue(failingLines.SetEquals(expectedLines),
-                "Expected failing lines do not match actual failed lines.  " +
-                $"Expected: [{string.Join(",", expectedLines)}]  Actual: [{string.Join(",", failingLines)}]");
+            var expectedLines = ruleExpectations.ReportedFailures.Select(failure => failure.LineNumber).ToList();
+            expectedLines.Sort();
+            Assert.IsTrue(failingLines.SequenceEqual(expectedLines),
+                "Expected failing lines do not match actual failed lines." + Environment.NewLine +
+                $"Expected: [{string.Join(",", expectedLines)}]  Actual: [{string.Join(",", failingLines)}]" +
+                (failingLines.Count > 0 ? "" : Environment.NewLine + "(Do the test directory and test config have the same name as the RuleId being tested?)"));
```

Comment:
Isn't `Do the test directory and test config have the same name as the RuleId being tested?` tested in L41? I don't see the pattern here why...

Reply:
Line 41 verifies the *template* specified in the config exists - the template can have any name.  This line reminds the user that the *name of the config* has to match the name of the rule.